src/elf.rs

```diff
@@ -275,21 +263,15 @@ impl<C: ContextObject> Executable<C> {
 
     /// Get the .text section virtual address and bytes
     pub fn get_text_bytes(&self) -> (u64, &[u8]) {
-        let (ro_offset, ro_section) = match &self.ro_section {
-            Section::Owned(offset, data) => (*offset, data.as_slice()),
+        let (offset, section) = match &self.text_section {
+            Section::Owned(_offset, _data) => unreachable!(),
             Section::Borrowed(offset, byte_range) => {
                 (*offset, &self.elf_bytes.as_slice()[byte_range.clone()])
             }
         };
-
-        let offset = self
-            .text_section_info
-            .vaddr
-            .saturating_sub(ebpf::MM_PROGRAM_START)
-            .saturating_sub(ro_offset as u64) as usize;
         (
-            self.text_section_info.vaddr,
-            &ro_section[offset..offset.saturating_add(self.text_section_info.offset_range.len())],
+            ebpf::MM_PROGRAM_START.saturating_add(offset as u64),
```

Comment:
I'm guesstimate that you are adding to `MM_PROGRAM_START` because you subtracted it from the offset in the initialization to maintain parallelism between this function and the `get_ro_region`, right? 

Would you mind explaining this in a comment?

Reply:
https://github.com/solana-labs/rbpf/blob/a400aa79a8b22c37b69311404d097246c97369ba/src/elf.rs#L237

The structs field is defined as being relative to `MM_PROGRAM_START`, both the text section and all read-only sections live in that `MemoryRegion`.

Reply:
In line 469, you're subtracting the values 

```
 text_section_vaddr.saturating_sub(ebpf::MM_PROGRAM_START) as usize,
```

My concern was that it was a little confusing that you subtract and then you add in `get_text_bytes`. A comment to explain why this is the case would be helpful.